File: models/src/main/java/ar/edu/itba/paw/model/AppointmentStatus.java
package ar.edu.itba.paw.model;

import ar.edu.itba.paw.model.exceptions.InvalidFilterException;

import java.util.Arrays;

public enum AppointmentStatus {
    CONFIRMED("confirmed"),
    PENDING("pending"),
    FINISHED("finished"),
    DENIED("denied"),
    CANCELLED("cancelled");

    private final String value;
    public static final String DEFAULT_VALUE = "pending";

    AppointmentStatus(final String value) {
        this.value = value;
    }

    public static AppointmentStatus toEnum(String value) {
        return Arrays.stream(values())
                .filter(s -> s.value.equalsIgnoreCase(value))
                .findFirst()
                .orElseThrow(InvalidFilterException::new);
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: frontend/src/models/enums/AppointmentStatus.ts
export enum AppointmentStatus {
    CONFIRMED = "confirmed",
    PENDING = "pending",
    DENIED = "denied",
    CANCELLED = "cancelled",
	FINISHED = "finished",
}

export function getAppointmentStatus(status: any, defaultStatus: AppointmentStatus): AppointmentStatus {
  return Object.values(AppointmentStatus).includes(status)
    ? status
    : defaultStatus;
}


export enum AppointmentView {
    BUSINESS= "businessId",
    USER= "userId"
} 